src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -73,7 +135,24 @@ private void submitRecordOperation(DataOperation op, OpType opType) {
 
   private void submitRecordDeleteOperation() {}
 
-  private void submitRecordReadOperation() {}
+  private void submitRecordReadOperation(final DataOperation recordOp, String nodeAddr) {
+    try {
+      val config = configCache.computeIfAbsent(nodeAddr, this::createConfig);
+      val adminConfig =
+          adminClientCreateFuncCache.computeIfAbsent(config, AdminClientCreateFunctionImpl::new);
+      val adminClient = adminClientCache.computeIfAbsent(nodeAddr, adminConfig);
+
+      val consConfig = consumerConfigCache.computeIfAbsent(nodeAddr, this::createConsumerConfig);
+      val consumerConfig =
+          consumerCreateFuncCache.computeIfAbsent(consConfig, ConsumerCreateFunctionImpl::new);
+      val kafkaConsumer = consumerCache.computeIfAbsent(nodeAddr, consumerConfig);
+
+      val result = kafkaConsumer.poll(Duration.ofSeconds(10));
```

Comment:
How should I process received records?

Reply:
happy path:
1. extract the record from the result
2. extract the record payload (value) size
3. call `countBytesDone` method for the operation instance
4. set the `SUCC` status for the operations instance